day2/src/main.rs

```diff
@@ -0,0 +1,58 @@
+use std::fs::read_to_string;
+
+fn main() {
+    let input = read_file("input.txt");
+    println!("Puzzle 1: {:?}", puzzle1(&input));
+    println!("Puzzle 2: {:?}", puzzle2(&input));
+}
+
+fn puzzle1(input: &str) -> u32 {
+    input
+        .split("\n")
+        .map(|line| {
+            let opponent = line.chars().nth(0).unwrap() as u32 - 'A' as u32;
+            let player = line.chars().nth(2).unwrap() as u32 - 'X' as u32;
+
+            game_score(player, opponent) + player + 1
+        })
+        .sum()
+}
+
+// computes the player's score for a round, given the player and opponent moves
+fn game_score(player: u32, opponent: u32) -> u32 {
+    if player == (opponent + 1) % 3 {
+        return 6;
+    }
+    if player == opponent {
+        return 3;
+    }
+    0
```

Comment:
Can also prefer to use `match` over `if-else`:

```suggestion
    match player {
        p if p == (opponent + 1) % 3 => 6,
        p if p == opponent => 3,
        _ => 0,
    }
```

See https://www.reddit.com/r/rust/comments/w625ur/is_there_any_advantage_of_using_match_instead_of/